### PR TITLE
New Advanced Build Your Own Shuttle Kit

### PR DESCRIPTION
**Pull request by Comxy**
## About The Pull Request
Adds a new emergency shuttle kit without any air and a few tiles in the middle. This way you can make your shuttle whatever way you want without having to deconstruct tiles to make gaps in your shuttle or needing the remove walls. You will also have to build your own thrusters on the shuttle. It costs 2000 credits (crate value x 10).
promo

## Why It's Good For The Game
Now the crew has better options for them to build their own shuttle.

## Changelog

:cl:
add: Adds a new emergency build kit shuttle
/:cl:

### CODE REVIEW

**Comment by Addust** [comment]
There should be lattice marking the exterior line if you don't have that, preferably with some structural connection to the middle tiles, which will need to have the console.

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -184,6 +184,18 @@
 	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/engineering/shuttle_engine]
 	P.special_enabled = TRUE
 
+/datum/map_template/shuttle/emergency/builder
+	suffix = "builder"
+	name = "Advanced build your own shuttle kit"
+	description = "Airless and tileless version of the build you own shuttle kit if you want to start building the shuttle from zero."
+	admin_notes = "Doesn't start out with air or tiles so it isn't very habitable."
+	credit_cost = CARGO_CRATE_VALUE * 10
```
Comment (JohnFulpWillard) [inline code/datums/shuttles.dm:192]: make the CE able to purchase it, like they can the default BYOS

Comment (Comxy) [reply code/datums/shuttles.dm:192]: Sure why not

**Comment by Toastgoats** [comment]
Probably a good idea to put screenshots and the overall credit cost in the pr, but otherwise I'm excited for this.

**Comment by D4C-420** [comment]
make sure to provide extra Eva suits incase someone buy's this and does no work on it.

**Comment by Toastgoats** [comment]
One more thing, it'd be nice to have a way to place the shuttle console manually instead of begging an admin to varedit it. Maybe a one-use beacon that lets you teleport it somewhere else in the shuttle?

**Comment by newfren** [comment]
this is just a grief shuttle

**Comment by Fikou** [comment]
This is way easier to get than shuttles like disco inferno or the meteor and its even more deadly (and cheap), its not fit for the roster, way too easy to grief with.

**Comment by Toastgoats** [comment]
Maybe have a living CE be a requirement for buying the shuttle? I wouldnt immediately jump to have it be a syndicate only shuttle but some strict requirement would help with the grief issue, aside from just bwoinking command for buying this just to grief.

**Comment by Toastgoats** [comment]
While you're at it, maybe even add a temporary shelter in the corner or something, like a mining pod.

**Comment by KommandantJackal** [comment]
I like this, just make it CE exclusive

**Comment by Toastgoats** [comment]
Another suggestion for anti-grief would be tying the shuttle's availability to a shipwrighting permit that's buyable from cargo.

**Comment by KommandantJackal** [comment]
I don't want this to be a station goal because I wanna build a BSA on it

**Comment by Toastgoats** [comment]
> I don't want this to be a station goal because I wanna build a BSA on it

Fantastic point.

**Comment by BlueMemesauce** [comment]
> I don't want this to be a station goal because I wanna build a BSA on it

you can unlock the BSA without it being station goal by using the keycard authorizer

**Comment by private-tristan** [comment]
please no, simply BUILDING the shuttle walls and giving it oxygen will take like, ~~20~~10 (saw the RCDs) minutes just to get to the point where you have a normal BYOS this is literally only going to be used for griefing or buying it to grab the RCD without needing to go to research as a tot

**Comment by Comxy** [comment]
I made is only available though the emag feature, if you guys still don't want this you can just close this PR.

**Comment by Toastgoats** [comment]
Having this be emag only prevents people from actually buying this in good faith, so it'd be nice if we could get a way to buy dangerous shuttles without an emag.

**Comment by distributivgesetz** [comment]
(crazy idea, probably not good)
Could you not check if it's a safe shuttle on a shuttle call (area check, breathable air), swap to a fallback if it isn't?

**Comment by Toastgoats** [comment]
I also just had another idea, but it's a longshot.

Rework this as a DIY _modular_ shuttle, hear me out.

It could be nearly the same, but spawns with unique bluespace pods representing the rooms you'd normally get with a shuttle, engineering/atmos, seating, the cockpit, etc.

That way it can be quickly set up if someone buys it in bad faith, but the pods are still optional if bought in good faith.

**Comment by GunHog** [comment]
Consider replacing some of those mats with a construction console (similar to the Aux Base). The base console could then be used to add the shuttle specific items (thrusters, place/move the shuttle computer, install brig tiles, etc). 

Even if the purchaser has no intent to build out the shuttle, its potential for grief is lessened by allowing someone to quickly set up the equipment, flooring, and walls during the launch timer.

**Comment by Mothblocks** [comment]
I don't want basically-the-old-BYOS-except-with-more-lattice no

**Pull request closed without merging**